build/NuGet.ruleset

```diff
@@ -7,6 +7,9 @@
     <Rule Id="CA2213" Action="Warning" />
     <Rule Id="CA2231" Action="Warning" />
   </Rules>
+  <Rules AnalyzerId="Microsoft.CodeAnalysis.CSharp.CodeStyle" RuleNamespace="Microsoft.CodeQuality.Analyzers">
+    <Rule Id="IDE0055" Action="Warning" />
+  </Rules>
```

Comment:
Give that IDE analyzers are only applicable in the context of VS, how about adding `dotnet_diagnostic.IDE0055.severity = warning` to the [`.editorconfig` file](https://github.com/NuGet/NuGet.Client/blob/dev/.editorconfig)?

I added `IDE0055` in my local branch `.editorconfig` file and VS reported a warning as expected.

![image](https://user-images.githubusercontent.com/52756182/124703976-26b1f180-dea8-11eb-836e-31d5bffe9308.png)

Reply:
There's a way to enforce them during build. I hadn't figured out why it wasn't working yet, but now I did. 
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview#code-style-analysis

Given that all of the build rules are there, I thought it'd be better to add this one there as well. 

Latest update has those changes.

Reply:
https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#enforcecodestyleinbuild doc says `EnforceCodeStyleInBuild` is still an experimental feature. I think we should be good.

I think we can we remove below task from the CI pipeline after this PR has been merged? The reason is IDE0055 analyzer will fail the build if a file requires formatting.

https://github.com/NuGet/NuGet.Client/blob/1d75910076b2ecfbe5f142227cfb4fb45c093a1e/eng/pipelines/templates/Build_and_UnitTest.yml#L186-L192

Reply:
Not yet, because NuGet.Commands cannot be enabled for automatic formatting due to the use of submodules. https://github.com/NuGet/NuGet.Client/pull/4138/files/49938aaa6ed7dd602bf0ad96125fc12884424ea1#diff-c8ae0d514975a391b2077e86b250c729770262d08a5ffe7eff53de5ff5f48c41